cpp/src/transform/transform.cpp

```diff
@@ -80,59 +86,80 @@ std::map<uint32_t, std::string> build_ptx_params(mutable_column_view output,
   return params;
 }
 
-std::vector<device_data_t> build_device_data(mutable_column_view output,
-                                             std::vector<column_view> const& inputs)
+cudf::jit::column_device_view to_jit_view(column_view const& view)
 {
-  std::vector<device_data_t> data;
+  return cudf::jit::column_device_view{
+    const_cast<void*>(view.head()), nullptr, view.null_mask(), view.type(), view.size()};
```

Comment:
Another consideration is that a `cudf::column_view` could be sliced which means it has a non-zero `offset` and represents only a partial view of the underlying data. This is handy in partitioning large columns with zero-copying. 
So I think the jit::column_device_view would need to be able to handle this kind of input.

Reply:
I've added that now